frontend/src/components/TableData/index.tsx

```diff
@@ -0,0 +1,190 @@
+import React from "react";
+import { Box, Button } from "@mui/material";
+import { Renderer } from "components/Renderer";
+import { Table } from "components/Table";
+import { useFormControls } from "hooks/useFormControls";
+import { useFormatTableData } from "hooks";
+import { useFormSubmit } from "hooks/useFormSubmit";
+import { GDXModal } from "components/GDXModal";
+import { apiAxios } from "utils";
+import { useQuery, UseQueryResult } from "react-query";
+import { FormikValues } from "formik";
+import { ReadForm } from "components/ReadForm";
+import { CreateForm } from "components/CreateForm";
+import { EditForm } from "components/EditForm";
+
+/* eslint "no-warning-comments": [1, { "terms": ["todo", "fixme"] }] */
+// todo: Define a good type. "Any" type temporarily permitted.
+export const TableData = ({
+  itemName,
+  tableName,
+  getOneUrl,
+  getAllUrl,
+  createFormInitialValues,
+  readFields,
+  editFields,
+}: {
+  itemName: string;
+  tableName: string;
+  getOneUrl: string;
+  getAllUrl: string;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  createFormInitialValues: any;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  readFields: any;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  editFields: any;
+}) => {
+  const {
+    handleEditMode,
+    handleOpen,
+    handleClose,
+    handleCurrentRowData,
+    handleFormType,
+    formType,
+    open,
+    editMode,
+    currentRowData,
+  } = useFormControls();
+
+  const { handlePost, handleUpdate, Notification } = useFormSubmit();
+
+  /**
+   * returns destructured props from the useFormatTableData hook.
+   *
+   * @param   {string}   tableName   - The name of the table that you are wanting data from.
+   * @param   {string}   apiEndPoint - The enpoint as which the API query will use for it's call.
+   * @param   {Function} handleClick - Function passed to the "view" button of the Table component.
+   * @returns {object}               {data, isLoading}  - "data" contains the columns and rows of data for your table.  isLoading is a boolean prop that changes to true if quering data and false if it has received the data.
+   */
+
+  const { data, isLoading } = useFormatTableData({
+    tableName: tableName,
+    apiEndPoint: getAllUrl,
+    handleClick: handleOpen,
+  });
+
+  /**
+   * getAmendment is the fetch function for react query to leverage.
```

Comment:
Minor: wrong function name in doc block.

Reply:
thank you, i'm not done with this code, but did want some feedback on it, fixed, will be included in next pr